### PR TITLE
feat(Kafka): kafka supports parameters updated

### PR DESCRIPTION
**Pull request by 346010762**
**What this PR does / why we need it**:
kafka supports parameters updated
**Special notes for your reviewer**:

**Release note**:

```release-note
kafka supports parameters updated
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dms/ TESTARGS='-run TestAccKafkaInstance'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccKafkaInstance -timeout 360m -parallel 4
=== RUN   TestAccKafkaInstancesDataSource_basic
=== PAUSE TestAccKafkaInstancesDataSource_basic
=== RUN   TestAccKafkaInstance_basic
=== PAUSE TestAccKafkaInstance_basic
=== RUN   TestAccKafkaInstance_prePaid
=== PAUSE TestAccKafkaInstance_prePaid
=== RUN   TestAccKafkaInstance_withEpsId
=== PAUSE TestAccKafkaInstance_withEpsId
=== RUN   TestAccKafkaInstance_compatible
=== PAUSE TestAccKafkaInstance_compatible
=== RUN   TestAccKafkaInstance_newFormat
=== PAUSE TestAccKafkaInstance_newFormat
=== CONT  TestAccKafkaInstancesDataSource_basic
=== CONT  TestAccKafkaInstance_withEpsId
=== CONT  TestAccKafkaInstance_newFormat
=== CONT  TestAccKafkaInstance_compatible
--- PASS: TestAccKafkaInstance_withEpsId (911.72s)
=== CONT  TestAccKafkaInstance_prePaid
--- PASS: TestAccKafkaInstancesDataSource_basic (945.88s)
=== CONT  TestAccKafkaInstance_basic
--- PASS: TestAccKafkaInstance_compatible (1069.03s)
--- PASS: TestAccKafkaInstance_prePaid (874.37s)
--- PASS: TestAccKafkaInstance_newFormat (2434.74s)
--- PASS: TestAccKafkaInstance_basic (1589.80s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       2535.766s

```

### CODE REVIEW

File: docs/resources/dms_kafka_instance.md [hunk 1]
```diff
@@ -219,6 +227,14 @@ The `cross_vpc_accesses` block supports:
 
 * `advertised_ip` - (Optional, String) The advertised IP Address or domain name.
 
+<a name="dms_parameters"></a>
+The `parameters` block supports:
+
+* `name` - (Required, String) Specifies the parameter name. Static parameter needs the instance to be restarted
+  to take effect.
```
Comment (houpeng80) [inline docs/resources/dms_kafka_instance.md]:
```suggestion
* `name` - (Required, String) Specifies the parameter name. Static parameter needs to restart the instance to take effect.
```

File: docs/resources/dms_kafka_instance.md [hunk 2]
```diff
@@ -194,6 +199,9 @@ The following arguments are supported:
   produced to or consumed from a topic that does not exist.
   The default value is false.
 
+* `parameters` - (Optional, List) Specify an array of one or more parameters to be set to the Kafka instance after
+  launched. You can check on console to see which parameters supported. Structure is documented below.
```
Comment (houpeng80) [inline docs/resources/dms_kafka_instance.md]:
```suggestion
* `parameters` - (Optional, List) Specifies the array of one or more parameters to be set to the Kafka instance after
  launched. The [parameters](#dms_parameters) structure is documented below.
```

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go [hunk 3]
```diff
@@ -1468,3 +1546,211 @@ func autoTopicTaskRefreshFunc(client *golangsdk.ServiceClient, instanceID string
...
+		RetryFunc:    retryFunc,
+		WaitFunc:     KafkaInstanceStateRefreshFunc(client, d.Id()),
+		WaitTarget:   []string{"RUNNING"},
+		Timeout:      d.Timeout(schema.TimeoutUpdate),
+		DelayTimeout: 10 * time.Second,
+		PollInterval: 10 * time.Second,
+	})
+	if err != nil && !handleTimeoutError(err) {
+		return fmt.Errorf("error modifying parameters for the Kafka instance (%s): %s", d.Id(), err)
+	}
```
Comment (houpeng80) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: the response will contain timeout? the method of handleTimeoutError may be unnecessary

File: docs/resources/dms_kafka_instance.md (hunk 1 above)
Comment (346010762) [reply docs/resources/dms_kafka_instance.md]: done

File: docs/resources/dms_kafka_instance.md (hunk 2 above)
Comment (346010762) [reply docs/resources/dms_kafka_instance.md]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go (hunk 3 above)
Comment (346010762) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: deleted the method.

**Pull request merged**